src/azul/service/manifest_service.py

```diff
@@ -883,15 +892,15 @@ def write_to(self, output: IO[str]) -> Optional[str]:
             #        https://github.com/DataBiosphere/azul/issues/2649
             for bundle in list(doc['bundles'])[0:100]:  # iterate over copy …
                 doc['bundles'] = [bundle]  # … to facilitate this in-place modification
-                row = {}
+                row = dict(file_url=self._azul_file_url(one(doc['contents']['files'])))
```

Comment:
Maybe it isn't cause for concern. But this asumes that doc has a `files` singleton (a correct assumption), could that ever change? That is what drives the singleton for this entity. Perhaps consider `s/files/self.entity_type()`, your call.

Reply:
I don't think that would actually help o make the code less brittle, since if `self.entity_type()` returns anything other than `"file"`, it's going to generate nonsensical URLs.